src/pages/dar_application/DataAccessRequest_new.js

```diff
@@ -0,0 +1,330 @@
+import { useState, useEffect, useCallback} from 'react';
+import { a, div, h, h4, span, p } from 'react-hyperscript-helpers';
+import isNil from 'lodash/fp/isNil';
+import { DataSet, DAR } from '../../libs/ajax';
+import { FormField, FormFieldTitle, FormFieldTypes, FormValidators } from '../../components/forms/forms';
+
+const searchOntologies = (query, callback) => {
+  let options = [];
+  DAR.getAutoCompleteOT(query).then(
+    items => {
+      options = items.map(function(item) {
+        return item.label;
+      });
+      callback(options);
+    });
+};
+
+const searchDatasets = (query, callback) => {
+  DataSet.searchDatasets(query).then(items => {
+    let options = items.map(function (item) {
+      return formatSearchDataset(item);
+    });
+    callback(options);
+  });
+};
+
+const formatSearchDataset = (ds) => {
+
+  return {
+    key: ds.dataSetId,
+    value: ds.dataSetId,
+    dataset: ds,
+    displayText: ds.datasetIdentifier,
+    label: span({}, [
+      span({style: {fontWeight: 'bold'}}, [ds.datasetIdentifier]), ' | ', ds.name]),
+  };
+};
+
+const datasetsContainDataUseFlag = (datasets, flag) => {
+  return datasets.some((ds) => ds?.dataUse[flag] === true);
+
+};
+
+const fetchAllDatasets = async (dsIds) => {
+  let datasets;
+  if (!isNil(dsIds)) {
+    datasets = await Promise.all(dsIds.map((id) => DataSet.getDatasetByIdV2(id)));
+  } else {
+    datasets = [];
+  }
+
+  return datasets;
+};
+
+export default function DataAccessRequest(props) {
+  const {
+    formFieldChange,
+    formData,
+    setIrbDocument,
+    setCollaborationLetter,
+    ariaLevel = 2
+  } = props;
+
+  const [datasets, setDatasets] = useState([]);
+  const today = new Date();
+  const irbProtocolExpiration = formData.irbProtocolExpiration || `${today.getFullYear().toString().padStart(4, '0')}-${today.getMonth().toString().padStart(2, '0')}-${today.getDate().toString().padStart(2, '0')}`;
+
+
+  useEffect(() => {
+    fetchAllDatasets(formData.datasetIds).then((datasets) => {
+      setDatasets(datasets);
+    });
+  }, [formData.datasetIds]);
+
+  const onChange = ({key, value}) => {
+    formFieldChange({name: key, value});
+  };
+
+  const needsIrbApprovalDocument = useCallback(() => {
+    return datasetsContainDataUseFlag(datasets, 'ethicsApprovalRequired');
+  }, [datasets]);
+
+  const needsCollaborationLetter = useCallback(() => {
+    return datasetsContainDataUseFlag(datasets, 'collaboratorRequired');
+  }, [datasets]);
+
+  const needsGsoAcknowledgement = useCallback(() => {
+    return datasetsContainDataUseFlag(datasets, 'gsoAcknowledgement');
+  }, [datasets]);
+
+  const needsPubAcknowledgement = useCallback(() => {
+    return datasetsContainDataUseFlag(datasets, 'publicationResults');
+  }, [datasets]);
+
+  const needsDsAcknowledgement = useCallback(() => {
+    return datasetsContainDataUseFlag(datasets, 'diseaseRestrictions');
```

Comment:
This check is a little different from the rest of the checks since the disease matters.

Let's say there are two datasets with the following DULs ...

Dataset A -> Disease Restrictions: study limited to cancer
Dataset B -> Disease Restrictions: study limited to atrial fibrillation

Both of the datasets have disease restrictions, but the diseases that they focus on makes each Data Use Limitation different from each other. 

Because of that, a request against these two datasets should trigger the question
`I acknowledge that the dataset can only be used in research consistent with the Data Use Limitations (DULs)...`.  Since you're only checking on whether `diseaseRestrictions` is true, that question will never populate.

In order to figure out if two `diseaseRestrictions` are equivalent to each other you'll need to compare the diseases that each dataset focuses on.

You can test this out by using DUOS-000045 and DUOS-000046 on dev (and local if you're using a copy of dev) when making drafts/requests

Reply:
wait - so is the checkbox rendered if there are any disease restrictions at all (array is not empty) or is there some other condition, (diseases match or do not match dar?)?

Reply:
The checkbox is rendered if any of the datasets have a Data Use Limitation (DUL) that is different from the other datasets selected. So let's assume...

Dataset A -> For general use only
Dataset B -> For studies regarding population, origins, and ancestry (POA)

Both of these have different DULs, so the checkbox has to pop up since its a prompt that forces the user to acknowledge that they will follow the restrictions for each dataset rather than just picking the least restrictive DUL (General Use in this case) and applying that across all the selected datasets.

Reply:
On top of that we need to consider that there is an implicit hierarchy for GRU, HMB, DS, and POA. Here's a diagram of the relationship below...

![1-s2 0-S2666979X21000355-gr1_lrg](https://user-images.githubusercontent.com/6664778/196682506-5b00c932-1829-45a4-affb-f2664ca80d18.jpg)

This comes into play when dealing with older datasets because their data is poorly formatted and may have multiple restrictions listed. 

So as an example, let's say a dataset has GRU, HMB, and DS listed in their data use restriction. Because DS is the most restrictive limitation, we evaluate that dataset as having only DS.

A second example: let's say that a dataset has both POA and DS selected. Because there's no nested relationship between the two, we can say that the dataset has both POA and DS as limitations.

There's also this edge case to consider: There exists a dataset with a DS restriction, but there's no diseases listed. While it doesn't make sense to say it has disease based restrictions, we can assume that there is a health/medical/biomedical slant to it, so we can treat it as an HMB.

Reply:
I think I understand. Here is the new code:

```
  useEffect(() => {
    translateDataUseRestrictionsFromDataUseArray(datasets.map((ds) => ds.dataUse)).then((translations) => {
      setDataUseTranslations(translations);
    });

  }, [datasets]);
```

```
  const needsDsAcknowledgement = useCallback(() => {
    // if any data use translations are different, then this must be displayed.
    return dataUseTranslations.length > 1 && !dataUseTranslations.reduceRight((prevValue, currValue) => isEqual(prevValue, currValue));
  }, [dataUseTranslations]);
```

So this new code will use ontology to translate the data uses properly into the heirarchy / structure that you described and then if there are more than one dataset, the checkbox is rendered if the data uses are not uniform. Is that the correct logic, now?

Reply:
Oh wow, that makes sense. I see why people like typescript now !! I simplified the logic (I was trying to be fancy but clearly I didn't understand fully how it worked). 

```
  const needsDsAcknowledgement = useCallback(() => {
    // if any data use translations are different, then this must be displayed.
    return dataUseTranslations.length > 1 && !dataUseTranslations.every((translation) => isEqual(dataUseTranslations[0], translation));
  }, [dataUseTranslations]);
```

New logic should be better. `DUOS-000005`, `DUOS-000007`, and `DUOS-000010` now do not render the checkbox.